doc/api/stream.md

```diff
@@ -154,89 +325,134 @@ Examples of readable streams include:
-explicitly paused will switch the stream into flowing mode. Data will
-then be passed as soon as it is available.
+The `'data'` event is emitted when the stream is in flowing mode and there is
+data available to be consumed. Attaching a `'data'` event listener to a stream 
+that has not been explicitly paused will switch the stream into flowing mode. 
+Data will then be passed as soon as it is available.
 
-If you just want to get all the data out of the stream as fast as
-possible, this is the best way to do so.
+The listener callback will be passed the chunk of data as a string if a default
+encoding has been specified for the stream using the 
+`readable.setEncoding()` method; otherwise the data will be passed as a
+`Buffer`.
+
+The `'data'` event is the most efficient means of consuming all of a Readable
+streams data:
 
 ```js
-var readable = getReadableStreamSomehow();
+const readable = getReadableStreamSomehow();
 readable.on('data', (chunk) => {
-  console.log('got %d bytes of data', chunk.length);
+  console.log(`Received ${chunk.length} bytes of data.`);
 });
 ```
 
-#### Event: 'end'
+##### Event: 'end'
 
-This event fires when there will be no more data to read.
+The `'end'` event is emitted when there is no more data to be consumed from
+the stream.
 
-Note that the `'end'` event **will not fire** unless the data is
-completely consumed. This can be done by switching into flowing mode,
-or by calling [`stream.read()`][stream-read] repeatedly until you get to the
-end.
+*Note*: The `'end'` event **will not be emitted** unless the data is
+completely consumed. This can be accomplished by switching the stream into
+flowing mode, or by calling [`stream.read()`][stream-read] repeatedly until
+all data has been consumed.
 
 ```js
-var readable = getReadableStreamSomehow();
+const readable = getReadableStreamSomehow();
 readable.on('data', (chunk) => {
-  console.log('got %d bytes of data', chunk.length);
+  console.log(`Received ${chunk.length} bytes of data.`);
 });
 readable.on('end', () => {
-  console.log('there will be no more data.');
+  console.log('There will be no more data.');
 });
 ```
 
-#### Event: 'error'
+##### Event: 'error'
 
 * {Error}
 
-Emitted if there was an error receiving data.
-
-#### Event: 'readable'
+The `'error'` event will be emitted if an error occurred while receiving data.
+The listener callback will be passed a single `Error` object.
 
-When a chunk of data can be read from the stream, it will emit a
-`'readable'` event.
+##### Event: 'readable'
 
-In some cases, listening for a `'readable'` event will cause some data
-to be read into the internal buffer from the underlying system, if it
-hadn't already.
+When the Readable stream is in paused mode, the `'readable'` event is emitted
+when there is data available to be read from the stream. In some cases,
+attaching a listener for the `'readable'` event will cause some data to be read
+into an internal buffer.
```

Comment:
Ooh, good catch, I had forgotten about this. On a fresh readable stream, if the `"readable"` event is listened for, we'll trigger a `.read(0)` if we're not already reading something. Effectively this triggers the data generation process _without_ triggering the data ownership transfer.